gr-blocks/include/gnuradio/blocks/wavfile_sink.h

```diff
@@ -35,12 +36,15 @@ class BLOCKS_API wavfile_sink : virtual public sync_block
      * \param filename The .wav file to be opened
      * \param n_channels Number of channels (2 = stereo or I/Q output)
      * \param sample_rate Sample rate [S/s]
-     * \param bits_per_sample 16 or 8 bit, default is 16
+     * \param format Output format (WAV, FLAC, Ogg Vorbis, RF64)
+     * \param subformat Bits per sample
+     * \param append Append to existing file
      */
     static sptr make(const char* filename,
                      int n_channels,
                      unsigned int sample_rate,
-                     int bits_per_sample = 16,
+                     wavfile_format_t format,
+                     wavfile_subformat_t subformat,
```

Comment:
Just an FYI -- I noticed this breaking API change when attempting to build Gqrx against 3.9.

https://github.com/csete/gqrx/blob/57334b493aac5a3c17aee909dba9c4349e02bee9/src/applications/gqrx/receiver.cpp#L1000

Gqrx can easily work around this with an `#ifdef`, as it already does in various places to maintain compatibility with multiple GNU Radio versions simultaneously. I just wanted to double check that the breaking change is intentional here. (It could presumably be avoided by overloading `make` but I'm not convinced it's worth doing that.)

Reply:
On second thought, the change to `std::shared_ptr` for blocks is already a much bigger API change in 3.9, so I don't think maintaining compatibility here is of any value.

Reply:
Yes, it's intentional. The WAV file sink and source in 3.9 have been upgraded to handle more file formats including FLAC and Ogg Vorbis (selected by wavfile_format_t) . Also, more WAV file bit depths (and float) are supported (selected by wavfile_subformat_t)..

It does require a new dependency, libsndfile1-dev. This should be resolved automatically in OOT's that include gnuradio-blocks.